### PR TITLE
Prettier readme

### PR DESCRIPTION
**Pull request by RebeccaStevens**
The current readme is too wordy - people don't like reading, they want to skim. This PR tries to trim down the readme to just the important information.

Also it adds a fancy header.

### CODE REVIEW

**Comment by RebeccaStevens** [comment]
@jonaskello Thoughts?

**Comment by jonaskello** [comment]
Yes, I agree this makes the readme shorter and nicer. It trims the reasoning for using immutability but that is probably not needed at this point. Nice touch with the logo too :-).
